hack/testing/test-viaq-data-model.py

```diff
@@ -119,10 +119,7 @@ def empty(thing):
               "undefined3": {"emptyvalue": ""}
     },
     "test7": {"myname":
-              {"undefined1": "undefined1",
-               "undefined11": 1111,
-               "undefined12": True,
-              },
+              "{\"undefined.6\":\"undefined6\",\"undefined1\":\"undefined1\",\"undefined11\":1111,\"undefined12\":true,\"undefined2\":{\"\":\"\",\"undefined2\":\"undefined2\",\"undefined22\":2222,\"undefined23\":false}}",
```

Comment:
This is going to be tricky - when the hash is converted back to a string, there is no guarantee of the order - so sometimes you will get the string `"{\"undefined.6\":\"undefined6\",\"undefined1\":\"undefined1\",...}"` and sometimes you will get the string `"{\"undefined1\":\"undefined1\",\"undefined.6\":\"undefined6\",...}"`

What I did in the go test for undefined to string is to convert the strings back to maps (which would be dicts in python), then compare the map (or dict) objects.